Nest/00. Plan.md

```diff
@@ -0,0 +1,67 @@
+# 1 Introduction
+## 1.1 Package manager
+
+## 1.2 Packages and dependencies
+
+## 1.3 Repositories
+
+## 1.4 Package's syntax
+
+# 2 Nest
+## 2.1 Nest's options
+
+## 2.2 Nest commands
+### 2.2.1 Synchronizing the local cache
+### 2.2.2 Troubleshooting
+
+## 2.3 Installing and uninstalling a package
+### 2.3.1 Installing a package
+### 2.3.2 Uninstalling a package
+### 2.3.3 Troubleshooting
+
+## 2.4 Updating your system
+### 2.4.1 Troubleshooting
+
+## 2.5 Conclusion
+
+# 3 Some more advanced concepts
+## 3.1 More about the repositories
+### 3.1.1 The official repositories
+### 3.1.2 The unofficial repositories
+### 3.1.3 Changing the repositories' configuration file
+
+## 3.2 The cache
+
+## 3.3 Version requirement
+
+# 4 Finest
+## 4.1 Key concepts
+### 4.1.1 The cache
+### 4.1.2 The transactions
+### 4.1.3 The dependency graph
+### 4.1.4 Groups
+### 4.1.5 Example
+
+## 4.2 Finest's options
+
+## 4.3 Finest's commands
+### 4.3.1 Installing and removing packages through requirements
+#### 4.3.1.1 Adding a requirement
+#### 4.3.3.2 Removing a requirement
+#### 4.3.3.3 Troubleshooting
+
+### 4.3.2 Updating your system by merging your dependency graph
+#### 4.3.2.1 Troubleshooting
+
+### 4.3.3 Actions on groups
+#### 4.3.3.1 Creating a group
+#### 4.3.3.2 Removing a group
```

Comment:
I think "Deleting a group" would better convey the meaning I imagine you're looking for.

Reply:
`Deleting` feels better, but the command is `finest remove`, so I felt like using the same verb was more relevant. I don't know if I should keep it like that or not though

Reply:
@Arignir Maybe finest needs to change in that case :^)

Reply:
The commands actually are `finest group add` and `finest group remove` (and not `finest group create` and `finest group remove` like the titles would suggest).

Surely that's something we can change, but the reason why they are name that way is to create a parallel with the requirements subcommands (`finest requirement add` and `finest requirement remove`)

What do @doom thinks of this?

Reply:
Obviously, the titles should match the subcommands.

Then, as to how these subcommands should be named, `remove` was picked to maintain homogeneousness with the other commands (`nest remove` and `finest requirement remove`), and because `removing a group` is actually `removing (a requirement to) the packages in this group`.
In the same way, `adding a group` is `creating a group and adding a requirement to it`, and not only `creating the group`.

The names / behaviors are of course open to debate if anyone wants to :smiley:

Reply:
I don't know, I was shocked that you were opposing create and remove, but if it's add and remove it seems ok for me.

Reply:
hmm I think I will simply change it as 'Creating a group' and 'Deleting a group'.
If I use 'add' and 'remove', I feel like `adding a group` sounds a bit weird (when `removing a group` doesn't), and `creating a group and adding a requirement to it` is a too long title :(

Reply:
I think it's important to respect the command names.

Reply:
Well I just realized that we decided to keep 3 levels of titles in the plan, and the ones we are arguing on are 4th level titles, so I think we don't care about those for the plan anymore :p